projects/Vula/test.nix

```diff
@@ -1,21 +1,81 @@
 {
   sources,
-  pkgs,
+  lib,
   ...
-}: {
+}: let
+  inherit (lib) recursiveUpdate;
+in {
   name = "vula";
 
-  nodes = {
-    server = {config, ...}: {
+  nodes = let
+    common = {
       imports = [
         sources.modules.default
         sources.modules."services.vula"
       ];
+
       services.vula.enable = true;
+      services.vula.openFirewall = true;
+
+      services.vula.logLevel = "DEBUG";
     };
+  in {
+    a = recursiveUpdate common {
+      services.vula.userPrefix = "fula";
+      services.vula.systemGroup = "zula";
+      services.vula.adminGroup = "vula-managers";
+      users.users.user.isNormalUser = true;
+      users.users.admin.isNormalUser = true;
+      users.users.admin.extraGroups = ["vula-managers"];
+    };
+
+    b = common;
   };
 
-  testScript = {nodes, ...}: ''
+  testScript = ''
     start_all()
+    a.wait_for_unit("vula-organize.service")
+    a.wait_for_unit("vula-publish.service")
+    a.wait_for_unit("vula-discover.service")
+    b.wait_for_unit("vula-organize.service")
+    b.wait_for_unit("vula-publish.service")
+    b.wait_for_unit("vula-discover.service")
+
+    def test_peer(node):
+        peer = 'a' if node.name == 'b' else 'b'
+        node.wait_until_succeeds(f"ping -I vula -w 1 {peer}", timeout=20)
+
+        peer_ip = node.succeed(f"getent hosts {peer}").split()[0]
+        route_result = node.succeed(f"ip route get {peer_ip}")
+        assert " dev vula " in route_result
+
+    test_peer(a)
+    test_peer(b)
+
+    a.succeed("pgrep --uid fula-organize vula")
```

Comment:
ideally vula-organize

Reply:
The default value is `vula-organize`. This is an assertion that a non-default value successfully applied.

Reply:
Understood, thank you for clarifying. It may be worth adding a comment to spell this out but for those who are nix experts, it may be absurd to add a comment. I don't mind either way and thank you for explaining it to me.